test/module/libs/common/combine_latest_until_first_completed_test.cpp

```diff
@@ -0,0 +1,239 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "common/combine_latest_until_first_completed.hpp"
+
+#include <chrono>
+
+#include <gtest/gtest.h>
+#include <boost/variant.hpp>
+#include "common/visitor.hpp"
+#include "rxcpp/rx.hpp"
+
+using namespace rxcpp;
+using namespace rxcpp::sources;
+using namespace rxcpp::subjects;
+using namespace rxcpp::util;
+using namespace std::literals::chrono_literals;
+using std::chrono::steady_clock;
+
+using time_point = steady_clock::time_point;
+using duration = std::chrono::milliseconds;
+
+struct IntervalObservableHelper {
+  IntervalObservableHelper(duration period)
+      : IntervalObservableHelper(steady_clock::now(), period) {}
+
+  IntervalObservableHelper(time_point start, duration period)
+      : observable_(rxcpp::observable<>::interval(start, period)),
+        start_(start),
+        period_(period) {}
+
+  virtual ~IntervalObservableHelper() = default;
+
+  long valueAtTime(time_point time) const {
+    return 1 + (time - start_) / period_;
+  }
+
+  long lastValue() const {
+    //return valueAtTime(steady_clock::now());
+    return last_emitted_value;
+  }
+
+  virtual rxcpp::observable<long> getObservable() {
+    return rememberLastVal(observable_);
+  }
+
+  rxcpp::observable<long> rememberLastVal(rxcpp::observable<long> observable) {
+    return observable.tap([this](long val) { this->last_emitted_value = val; });
+  }
+
+  rxcpp::observable<long> observable_;
+  time_point start_;
+  duration period_;
+  long last_emitted_value{0};
+};
+
+struct IntervalObservebleTakeNHelper : public IntervalObservableHelper {
```

Comment:
Why not to fix all the occurrences of the typo? :)

```suggestion
struct IntervalObservableTakeNHelper : public IntervalObservableHelper {
```